test/erl_optics_test_model.erl

```diff
@@ -97,17 +110,29 @@ percentile(Lst, N) ->
 
 populate_histo(Events, Buckets) ->
     Map1 = #{above => 0, below => 0},
-    Keys = lists:sort(Buckets),
+    Bucket_tuples = buckets_to_tuples(Buckets),
+    Keys = lists:sort(Bucket_tuples),
     Map2 = lists:foldl(fun(Bucket, Acc) -> Acc#{Bucket => 0} end, Map1, Keys),
     Map3 = lists:foldl(fun(Event, Acc) ->
         Key = determine_histo_key(Event, Keys),
         #{Key := Val} = Acc,
         Acc#{Key => Val + 1}
     end, Map2, Events),
-    LastBucket = lists:last(Buckets),
-    {V, Map4 = #{above := Above}} = maps:take(LastBucket, Map3),
-    Map4#{above => Above + V}.
+    Map3.
+
+
+buckets_to_tuples(Buckets)->
+    Sorted_buckets = lists:sort(Buckets),
+    buckets_to_tuples(Sorted_buckets, []).
 
+buckets_to_tuples([], Acc) ->
+    Acc;
+buckets_to_tuples([_|[]], Acc) ->
```

Comment:
The pattern here can be `[_]` instead. It looks like you're trying to match on a list with one element.